controllers/dpa_controller.go

```diff
@@ -136,14 +139,69 @@ func (r *DPAReconciler) SetupWithManager(mgr ctrl.Manager) error {
 		Owns(&velerov1.VolumeSnapshotLocation{}).
 		Owns(&appsv1.DaemonSet{}).
 		Owns(&security.SecurityContextConstraints{}).
-		Owns(&corev1.Secret{}).
 		Owns(&corev1.Service{}).
 		Owns(&routev1.Route{}).
 		Owns(&corev1.ConfigMap{}).
+		Watches(&source.Kind{Type: &corev1.Secret{}}, &labelHandler{}).
 		WithEventFilter(veleroPredicate(r.Scheme)).
 		Complete(r)
 }
 
+type labelHandler struct {
+}
+
+func (l *labelHandler) Create(evt event.CreateEvent, q workqueue.RateLimitingInterface) {
+	// check for the label & add it to the queue
+
+	if evt.Object.GetLabels()[oadpv1alpha1.OadpOperatorLabel] == "" {
+		return
+	}
+	dpaname := evt.Object.GetLabels()["dataprotectionapplication"]
```

Comment:
handle the case where dpaname is not set.

Reply:
done